mlrun/model_monitoring/db/stores/v3io_kv/kv_store.py

```
@@ -703,3 +702,57 @@ def get_v3io_monitoring_apps_container(project_name: str) -> str:
+        items: list[dict[str, str]],
+    ) -> list[mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric]:
+        metrics: list[
+            mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric
+        ] = []
+        for app_item in items:
+            # See https://www.iguazio.com/docs/latest-release/services/data-layer/reference/system-attributes/#sys-attr-__name
+            app_name = app_item.pop("__name")
+            if app_name == ".#schema":
+                continue
+            for result_name in app_item:
+                metrics.append(
+                    mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric(
+                        app=app_name,
+                        name=result_name,
+                        type=mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetricType.RESULT,
+                        full_name=f"{app_name}/{mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetricType.RESULT}/{result_name}",
+                    )
+                )
+        return metrics
+
+    def get_model_endpoint_metrics(
+        self, endpoint_id: str
+    ) -> list[mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric]:
+        """Get model monitoring results and metrics on the endpoint"""
+        metrics: list[
+            mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric
+        ] = []
+        container = self.get_v3io_monitoring_apps_container(self.project)
+        try:
+            response = self.client.kv.scan(container=container, table_path=endpoint_id)
+        except v3io.dataplane.response.HttpResponseError:
+            logger.info(
+                "Attempt getting metrics and results - no data. Check the "
+                "project name, endpoint, or wait for the applications to start.",
+                container=container,
+                table_path=endpoint_id,
+            )
+            return []
```

This has the strange effect that if the first call fails, we log an info message and return an empty list, but if subsequent calls fail, we raise an error.

---

That's right.
However, if we pass that try-except block, then the table for sure exists, and the error is caused by a different source.

---

Sure (if you assume no one will delete the table concurrently), but this block will silence all errors.

---

Fixed to suppressing only 404 in 4744fca